src/app/pages/signup-page/signup-page.component.html

```diff
@@ -1 +1,68 @@
-<p>signup-page works!</p>
+<h2>SighUp</h2>
+
+<form [formGroup]="registryForm" (ngSubmit)="onSubmit()" [ngClass]="firstContact ? 'active' : null">
+  <mat-form-field appearance="outline">
+    <mat-label for="email">Email</mat-label>
+    <input matInput type="text" id="email" formControlName="email" (keydown)="onClearError()" />
+    <mat-error *ngIf="email.hasError('required') && email.dirty && email.touched && firstContact">
+      Required</mat-error
+    >
+    <mat-error
+      *ngIf="
+        email.errors?.['EmailError'] &&
+        email.dirty &&
+        email.touched &&
+        email.value.length &&
+        firstContact
+      "
+    >
+      Incorrect email
+    </mat-error>
+  </mat-form-field>
+  <div [ngClass]="(reason$ | async) === 'invalidUniqueKey' ? 'error' : 'hide'">
+    Account with this email already exists
+  </div>
+  <mat-form-field appearance="outline">
+    <mat-label for="password">Password</mat-label>
+    <input matInput type="password" id="password" formControlName="password" />
+    <mat-error
+      *ngIf="password.hasError('required') && password.dirty && password.touched && firstContact"
+    >
+      Required</mat-error
+    >
+    <mat-error
```

Comment:
also you can replace it in component method and save space

Reply:
this place I fixed 3 times and it change again and again)))